src/services/lostAndFound.ts

```diff
@@ -0,0 +1,138 @@
+import http from './http';
+import userService from './user';
+
+/**
+ * Missing item report object, representing the model of a report for communication with the
+ * backend service.
+ */
+export type MissingItemReport = {
+  recordID?: number;
```

Comment:
Why is `recordID` nullable?

Reply:
If the type of the objects is different, we should use different types for them. I see that you're actually using the type `Omit<MissingItemReport, 'recordID'>` when posting, which is the correct approach. You can save that as a new type, e.g. 
```typescript
export type NewMissingItemReport = Omit<MissingItemReport, 'recordID'>
```

But if the value returned from the API never has a null/undefined `recordID`, we should model that property as never being null/undefined. Otherwise, you have to fight with typescript downstream because you know that it's defined but typescript doesn't and will complain when you try to access it.

The same thing applies to the ID fields on the other objects you model in this file.